### ISSUE TITLE
Use drupal-project .gitignore for composer-based Drupal projects

### ISSUE THREAD

**Issue by joshuaboltzmc** [ironstar-io/tokaido#195]
**Is your feature request related to a problem? Please describe.**
Yes, because when you initialize a new Tokaido site, you end up with a .gitignore that looks like this
```
# START Generated by Tokaido
docker-compose.tok.yml
.tok/local
private/default/*
web/sites/*/settings.tok.php
web/sites/*/files/*

# END Generated by Tokaido
```
Tokaido also goes ahead and makes the initial commit to the repo, and contained in that commit are Tokaido configurations, but also various other files, such as /vendor files that in a composer-based Drupal setup, we don't want tracked in Git.

So, the next sensible step is to add .gitignore values to block files and folders like /vendor, /core, /modules/contrib from being tracked in Git.

To do this, I can pull most of this into my .gitignore:
https://github.com/drupal-composer/drupal-project/blob/8.x/.gitignore

But, after making a change to composer, such as pulling in a new contrib module, I still see the vendor files being tracked with Git.

**Describe the solution you'd like**
The .gitignore provided by Tokaido should contain the proper .gitignore to keep this from happening. In the case of a non-composer-based Drupal project (are there any with Drupal 8), maybe one of the steps during `tok new` is a question asking `Is this a composer based site?`? If so, update the .gitignore to exclude vendor, core, etc.

**Describe alternatives you've considered**
I can run this command `$ git rm -r --cached vendor`
which does some cleanup and then results in those vendor files no longer being tracked, but I feel like this is something Tokaido can add for us. Just seems like we have to work around something that should have been done automatically for us, and could be a hinderance to people trying to get up and running and not knowing the `$ git rm -r --cached vendor` trick.

**Comment by otakumike** [comment]
Thanks for the issue @joshuaboltzmc!

@yjimk what are your thoughts on this? Would it be better to modify the standard .gitignore a bit more, or maybe just leave out that initial commit and leave that to the user?

**Comment by garrettw** [comment]
`git rm -r --cached vendor` << This is correct and not really a "trick" if you understand how gitignore works. Gitignore only controls how git handles untracked files; once they're tracked, that won't keep git from seeing changes.

Asking `Is this a composer based site?` might could be avoided by simply sniffing for the existence of a `composer.json`.

**Comment by otakumike** [comment]
Tokaido is designed to only work with composer, so we don't have to worry about checking for composer or not. 

Some hosts want you to commit the vendor directory when you use their remote Git repo (for example, lots of Acquia sites just commit /vendor) because it's easier than having an external CI or having to pay for Pipelines. So there are use cases where people really want to commit vendor, and if we change this functionality suddenly their `tok new` sites won't work. 

I definitely think the best course of action here is to just not do the initial Git commit and leave it to the user to decide when they're ready to do that. I do like the cleanliness of `tok new` giving you a fresh site all committed and tidy, but we _are_ making assumptions there that we probably shouldn't.

**Comment by yjimk** [comment]
Thanks for the GH issue @joshuaboltzmc! It raises a good question that should be addressed.

I'm not sure changing the action of `tok new` to not create a commit entirely solves the problem. IMO both the change to `tok new` and additional `.gitignore` fields should be considered.

My reasoning for this is that Tokaido is a relatively opinionated piece of software, we're trying to edge people toward what we would consider "best practice" for a Drupal project. Committing vendor/node_modules etc. isn't best practice, even if it is unavoidable in some circumstances. 

The other point to this is that in git, it is much easier and more intuitive to add a file/directory for tracking than it is to remove it.

In summary I'd propose the following changes
- No auto-commit on `tok new`
- Extended list of files/directories in `.gitignore`

What does everyone think?

**Comment by otakumike** [comment]
> No auto-commit on `tok new`

Agreed, this makes sense

> Extended list of files/directories in `.gitignore`

Also agree, but this should only be done by `tok new`, when adding stuff to `.gitignore` as part of a `tok up` operation, I don't think we should change the `.gitignore` file and more than we already are. The user at this point in an existing project would most likely have already decided what they want to do with /vendor

I'll mark this to be added to v1.11

**Comment by otakumike** [comment]
This is mostly fixed now with the inclusion of a .gitignore file in new templates (which works from now). 

But this PR just merged will stop the auto-commit to Git, but still stages all of the auto-generated stuff.
